### ISSUE TITLE
Plugin does not work with stylelint^9.2.0

### ISSUE THREAD

**Issue by sobolevn** [AndyOGo/stylelint-declaration-strict-value#22]
I am having trouble integrating your plugin to my configuration: https://github.com/wemake-services/stylelint-config-strict-scss/issues/18

When installing your plugin like so:

```js
  plugins: [
    ...
    'stylelint-declaration-strict-value'
  ],

  rules: [
    // use variables for these properties:
    'scale-unlimited/declaration-strict-value': [
      ['/color/', 'z-index', 'font-size'],
      {
        ignoreKeywords: {
          '': ['inherit'],
          '/color/': ['currentColor', 'transparent', 'inherit']
        }
      }
    ],
  ]
```

Every test fails with the following message:

```
  ● if-else tags › incorrect style › flags correct number of warnings

    stylelint v3+ requires plugins to expose a ruleName. The plugin "/Users/sobolev/Documents/github/stylelint-config-standard-scss/node_modules/stylelint-declaration-strict-value/index.js" is not doing this, so will not work with stylelint v3+. Please file an issue with the plugin.

      at Object.<anonymous>.module.exports (node_modules/stylelint/lib/utils/configurationError.js:8:28)
      at normalizedPluginImport.forEach.pluginRuleDefinition (node_modules/stylelint/lib/augmentConfig.js:264:15)
          at Array.forEach (<anonymous>)
      at normalizedPlugins.reduce (node_modules/stylelint/lib/augmentConfig.js:262:28)
          at Array.reduce (<anonymous>)
      at addPluginFunctions (node_modules/stylelint/lib/augmentConfig.js:251:45)
      at augmentConfigBasic.then.augmentedConfig (node_modules/stylelint/lib/augmentConfig.js:77:14)
```

**Comment by AndyOGo** [comment]
@sobolevn 
Thanks for your report. Though isn't a bug here.
I export the `ruleName` and  I knowthis plugin works in stylelint `7`, `8` (I personally used it there).

Here is the code, where `ruleName` is exported:
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/src/index.js#L128
```js
export { ruleName, messages }
```

**Comment by AndyOGo** [comment]
PS: please file an issue over at stylelint

**Comment by sobolevn** [comment]
@AndyOGo ok, here's the thing.

Your plugin actually works with the latest `stylelint`. I have manually tested it.
But the tests fail. 

Here's how I test my configuration:

```js
const path = require('path')
const stylelint = require('stylelint')

...

   beforeAll(() => {
      result = stylelint.lint({
        files: path.join(fixturesDir, name, 'incorrect.scss'),
        config
      })
    })

    test('flags correct number of warnings', async () => {
      expect.hasAssertions()
      expect.assertions(2)

      const data = await result

      expect(data.results[0].warnings).toHaveLength(rule.errors)
      expect(data.errored).toBeTruthy()
    })
```

## Env

I am using `jest` as a test runner.

Things I have tried:

- using `babel-jest` as a transformer for tests
- switching to `import stylelint from 'stylelint'`

## Problem

When running tests these two values are empty: https://github.com/stylelint/stylelint/blob/master/lib/augmentConfig.js#L262-L263

Output:

```
  console.log node_modules/stylelint/lib/augmentConfig.js:262
    import [ {} ]

  console.log node_modules/stylelint/lib/augmentConfig.js:264
    instance {}
```

Trying to investigate further.

**Comment by sobolevn** [comment]
Here's the difference between using CLI:

```
normalized {} /Users/sobolev/Documents/github/stylelint-config-standard-scss/node_modules/stylelint-declaration-strict-value/index.js
dynamic { ruleName: 'scale-unlimited/declaration-strict-value',
  messages: { expected: [Function] },
  default: 
   { ruleName: 'scale-unlimited/declaration-strict-value',
     rule: { [Function: rule] primaryOptionArray: true } } }
es5/6 { ruleName: 'scale-unlimited/declaration-strict-value',
  rule: { [Function: rule] primaryOptionArray: true } }
```

And running tests:

```
Ran all test suites.
  console.log node_modules/stylelint/lib/augmentConfig.js:252
    normalized {} /Users/sobolev/Documents/github/stylelint-config-standard-scss/node_modules/stylelint-declaration-strict-value/index.js

  console.log node_modules/stylelint/lib/augmentConfig.js:255
    dynamic { ruleName: undefined, messages: undefined }

  console.log node_modules/stylelint/lib/augmentConfig.js:257
    es5/6 { ruleName: undefined, messages: undefined }
```

Here are the console prints:

```js
    console.log('normalized', result, pluginLookup)
    let pluginImport = dynamicRequire(pluginLookup);
    // Handle either ES6 or CommonJS modules
    console.log('dynamic', pluginImport)
    pluginImport = pluginImport.default || pluginImport;
    console.log('es5/6', pluginImport)
```

**Comment by AndyOGo** [comment]
@sobolevn 
Very interesting issue and really great and detailed bug report - I appreciate that.

Regarding my build and the resulting module format I can only show you that I use a straight-forward babel-only transpilation process via an npm script called `build`:
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/package.json#L7

May `.babelrc` file says the details.
https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/.babelrc

I think it's definitely time to provide a micheal jackson formatted version.
Will track that in another issue though.
